includes/Tracking/ConversionTrackingService.php

```diff
@@ -172,9 +172,12 @@ public function handle_single_product_add_to_cart( string $cart_item_key, int $p
 	public function handle_async_add_to_cart(): void {
 		check_ajax_referer( 'capi_nonce', 'security' );
 
-		$product_id = absint( wp_unslash( $_POST['product_id'] ?? 0 ) );
-		$quantity   = absint( wp_unslash( $_POST['quantity'] ?? 0 ) );
-		$event_id   = sanitize_text_field( wp_unslash( $_POST['event_id'] ?? '' ) );
+		$raw_input  = filter_input( INPUT_POST, 'payload', FILTER_UNSAFE_RAW );
+		$raw_input  = wp_unslash( $raw_input );
+		$data       = json_decode( $raw_input, true );
+		$product_id = isset( $data['product_id'] ) ? absint( $data['product_id'] ) : 0;
+		$quantity   = isset( $data['quantity'] ) ? absint( $data['quantity'] ) : 0;
+		$event_id   = isset( $data['event_id'] ) ? sanitize_text_field( $data['event_id'] ) : '';
```

Comment:
Any reason we moved from using the null coalescing operator here? Seems we could have kept that for simplicity
```suggestion
		$product_id = absint( $data['product_id'] ?? 0 );
		$quantity   = absint( $data['quantity'] ?? 0 );
		$event_id   = sanitize_text_field( $data['event_id'] ?? '' );
```